pkg/cloudprovider/cloud.go

```diff
@@ -66,6 +66,16 @@ type Clusters interface {
 // TODO(#6812): Use a shorter name that's less likely to be longer than cloud
 // providers' name length limits.
 func GetLoadBalancerName(service *v1.Service) string {
+        // Check if annotation for custom loadbalancer name is defined
+        lbname, ok := service.Annotations["load-balancer-name"]
```

Comment:
This should be something like `kubernetes.io/load-balancer-name`

Reply:
Now that I think about it a little more, I'm not sure how we would handle naming collisions since most providers lookup load balancers by name. If you pushed two services with the same name, then they'd fight. We'd need to use the namespace + service name, which still has problems when you run multiple clusters.